### ISSUE TITLE
Uploads under Xamarin/iOS are corrupted and invalid

### ISSUE THREAD

**Issue by kiddailey** [gpailler/MegaApiClient#152]
**Description**:

When uploading a stream or file under Xamarin/iOS, the file upload completes but is corrupted.  This can be confirmed by two things:

* The filename displays as "Undecrypted File" in the MEGA Cloud file listing
* Attempting to download the file via the API or MEGA file list directly fails with an invalid key error

The byte size reported by MEGA does appear correct, but for whatever reason, the encryption is failing.  I am trying to figure out where the issue is occurring, but my best guess so far is that it is has something to do with _MegaAesCtrStreamCrypter_ or deserialization.  Will update issue as I find out additional information.

**Steps to Reproduce**:

The following sample code running under Xamarin/iOS will reproduce the error.  Note that you will need to add NSAppTransportSecurity settings to your info.plist (see https://github.com/gpailler/MegaApiClient/issues/150#issuecomment-669399158 ) in order for the Upload() method to work successfully.

```c#
      var apiClient = new MegaApiClient();
      apiClient.Login("test@example.com", "abc123");

      IEnumerable<INode> nodes = apiClient.GetNodes();
      var rootNode = nodes.Single(x => x.Type == NodeType.Root);

      var ms = "This is a test";
      apiClient.Upload(new MemoryStream(Encoding.ASCII.GetBytes(ms)), "test.txt", rootNode);
      apiClient.Logout();
```

**MegaApiClient Version**: 1.8.1

**Notes**

* While stepping through upload on iOS, I'm seeing what I think is an unexpected result in the final createNodeRequest at the end of the upload.  Wen examining the returned node from mega, the name attribute contains the below error.  When stepping through in a Windows console app, this error does not occur.
     > "Attribute deserialization failed: Unexpected character encountered while parsing value: s. Path '', …"
* Pasting the exact same code into a Windows console app uploads the file correctly without issue
* I've confirmed that the masterKey is correct on both console and iOS traces
* It appears that iOS fails to deserialize responses properly for some reason.  Specifically, in RequestCore<TResponse> the last line performs a deserialization that appears to fail on iOS.
     > return (typeof(TResponse) == typeof(string)) ? data as TResponse : JsonConvert.DeserializeObject<TResponse>(data, new GetNodesResponseConverter(key));

* The attribute deserialization failure happens in ReadJSON as the serializer.Populate(jObjectReader, target) is executed, and when   OnDeserialized fires for GetNodesResponse
* **The actual exception is being thrown by Crypto.cs in the DecryptAttributes() method, line 152.  This is due to the decryptedAttributes not being decrypted properly and instead are random bytes instead of valid JSON.**  This appears to be where the actual issue is occurring.  Need to determine why the nodeKey and/or attribute bytes passed into this are not correct under iOS.

**Comment by kiddailey** [comment]
Yes, indeed it was an encryption problem and looks like it boils down to to the differences between using Aes.Create() and AesManaged() from System.Security.Cryptography.  There is a compiler conditional in Crypto.cs' constructor:

```c#
    static Crypto()
    {
#if NETSTANDARD1_3
      AesCbc = Aes.Create(); // More per-call overhead but supported everywhere.
      IsKnownReusable = false;
#else
      AesCbc = new AesManaged();
      IsKnownReusable = true;
#endif

      AesCbc.Padding = PaddingMode.None;
      AesCbc.Mode = CipherMode.CBC;
    }
```

My Xamarin project is NetStandard 2.0, and modifying this so that it also uses Aes.Create() for 2.0 resolves the issue.

```c#
    static Crypto()
    {
#if NETSTANDARD1_3 || NETSTANDARD2_0
      AesCbc = Aes.Create(); // More per-call overhead but supported everywhere.
      IsKnownReusable = false;
#else
      AesCbc = new AesManaged();
      IsKnownReusable = true;
#endif

      AesCbc.Padding = PaddingMode.None;
      AesCbc.Mode = CipherMode.CBC;
    }
```

It is possible that it could also perhaps be a CipherMode problem as I've seen that in other encryption cases with Xamarin/NetStandard projects, but I didn't experiment further since this fix seems sufficient for my needs.  I'm going to continue testing to ensure that this resolves all issues and if so, may submit a pull request.  Thanks!

**Comment by gpailler** [comment]
Issue fixed on 1.8.2
